src/ApiCore/GapicClientTrait.php

```diff
@@ -96,105 +97,165 @@ private static function getGapicVersion(array $options)
-     *           ['retriesEnabled' => false]. Retry settings provided in this setting override the
-     *           settings in $clientConfigPath.
-     *     @type callable $authHttpHandler A handler used to deliver PSR-7
-     *           requests specifically for authentication. Should match a
-     *           signature of
-     *           `function (RequestInterface $request, array $options) : ResponseInterface`.
-     *     @type callable $httpHandler A handler used to deliver PSR-7 requests.
-     *           Should match a signature of
-     *           `function (RequestInterface $request, array $options) : PromiseInterface`.
-     *           NOTE: This option is only valid when utilizing the REST transport.
+     *           If the $clientConfig option is specified, this will be ignored.
+     *     @type bool $disableRetries
+     *           Determines whether or not retries defined by the client configuration should be
+     *           disabled. Defaults to `false`.
      *     @type string|TransportInterface $transport
      *           The transport used for executing network requests. May be either
      *           the string `rest` or `grpc`. Additionally, it is possible
      *           to pass in an already instantiated transport. Defaults to `grpc`
      *           if gRPC support is detected on the system.
+     *     @type CacheItemPoolInterface $authCache A cache for storing access
+     *           tokens. Defaults to a simple in memory implementation.
      *     @type string $libName
      *           The name of the client application.
      *     @type string $libVersion
      *           The version of the client application.
      *     @type string $gapicVersion
      *           The code generator version of the GAPIC library.
      * }
+     * @throws ValidationException
      */
     protected function setClientOptions(array $options)
+    {
+        $this->validateOptions($options);
+
+        $this->setServiceNameAndDescriptors($options);
+        $this->setRetrySettings($options);
+        $this->setAgentHeaderDescriptor($options);
+        $this->setTransport($options);
+    }
+
+    /**
+     * @param array $options
+     * @throws ValidationException
+     */
+    private function validateOptions(array $options)
     {
         $this->validateNotNull($options, [
             'serviceName',
+            'serviceAddress',
             'descriptorsConfigPath',
-            'clientConfigPath'
+            'clientConfigPath',
+            'scopes',
         ]);
-        $transport = isset($options['transport'])
-            ? $options['transport']
-            : null;
-        $clientConfig = json_decode(
-            file_get_contents($options['clientConfigPath']),
-            true
-        );
-        $this->serviceName = $options['serviceName'];
-        $this->retrySettings = RetrySettings::load(
-            $this->serviceName,
+
+        if (isset($options['transport'])) {
```

Comment:
I believe the same should apply here as well. Does it seem we should worry about validating the transport in the factory, as this just gets passed off there already?